core/src/mast/serialization/mod.rs

```diff
@@ -125,38 +122,47 @@ impl Serializable for MastForest {
                     after_exit_decorators.push((mast_node_id, mast_node.after_exit().to_vec()));
                 }
 
-                let (ops_offset, decorator_data_offset) = if let MastNode::Block(basic_block) =
-                    mast_node
-                {
-                    let (ops_offset, decorator_data_offset) =
-                        basic_block_data_builder.encode_basic_block(basic_block);
+                let ops_offset = if let MastNode::Block(basic_block) = mast_node {
+                    let ops_offset = basic_block_data_builder.encode_basic_block(basic_block);
 
-                    (ops_offset, decorator_data_offset.unwrap_or(MastForest::MAX_DECORATORS as u32))
+                    basic_block_decorators.push((mast_node_id, basic_block.decorators().clone()));
+
+                    ops_offset
                 } else {
-                    (0, 0)
+                    0
                 };
 
-                MastNodeInfo::new(mast_node, ops_offset, decorator_data_offset)
+                MastNodeInfo::new(mast_node, ops_offset)
             })
             .collect();
 
-        let decorator_data = decorator_data_builder.finalize();
         let node_data = basic_block_data_builder.finalize();
-        let string_table = string_table_builder.into_table();
+        node_data.write_into(target);
```

Comment:
```suggestion
        basic_block_data.write_into(target);
```
Nit: We should align the name with the serialization format description at the beginning of the file.